### ISSUE TITLE
dotnet new3 --help results in NullRef

### ISSUE THREAD

**Issue by piotrpMSFT** [dotnet/cli#5402]
## Steps to reproduce
`dotnet new3 --help`

## Expected  behavior
Help dialog

## Actual behavior
```
$ dotnet new3 --help
Template Instantiation Commands for .NET Core CLI.

Usage: dotnet new3 [arguments] [options]

Arguments:
  template  The template to instantiate.

Options:
  -l|--list         List templates containing the specified name.
  -lang|--language  Specifies the language of the template to create
  -n|--name         The name for the output being created. If no name is specified, the name of the current directory is used.
  -o|--output       Location to place the generated output.
  -h|--help         Displays help for this command.
  -all|--show-all   Shows all templates

Templates         Short Name       Language      Tags
-------------------------------------------------------
Nuget Config      nugetconfig                    Config

Examples:
Object reference not set to an instance of an object.
   at Microsoft.TemplateEngine.Cli.New3Command.GenerateUsageForTemplate(ITemplateInfo templateInfo)
   at Microsoft.TemplateEngine.Cli.New3Command.ShowInvocationExamples()
   at Microsoft.TemplateEngine.Cli.New3Command.DisplayTemplateList()
   at Microsoft.TemplateEngine.Cli.New3Command.<EnterMaintenanceFlowAsync>d__55.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.TemplateEngine.Cli.New3Command.<ExecuteAsync>d__58.MoveNext()
```

## Environment data
.NET Command Line Tools (1.0.0-rc4-004536)

Product Information:
 Version:            1.0.0-rc4-004536
 Commit SHA-1 hash:  f297144c48

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.11-x64

**Comment by mlorbetske** [comment]
@seancpeters let's try this out

**Comment by seancpeters** [comment]
I'm unable to reproduce this behavior, but I'm testing on a windows machine. This error did exist at some point in the new3 code, but it was updated quite a while ago (many weeks).

**Comment by piotrpMSFT** [comment]
@mlorbetske I know you have a fix for this. When is it coming into CLI?

**Comment by piotrpMSFT** [comment]
Fixed by #5430
